connectors/gitlab/element-templates/gitlab-connector.json

```diff
@@ -852,66 +889,562 @@
       }
     },
     {
-      "label": "Release name",
-      "description": "The release name",
+      "label": "Search string",
+      "description": "Return list of branches containing the search string",
       "group": "operation",
       "type": "String",
       "feel": "optional",
       "optional": true,
       "binding": {
         "type": "zeebe:input",
-        "name": "releaseName"
+        "name": "searchString"
       },
       "condition": {
         "property": "operation",
         "oneOf": [
-          "createRelease"
+          "getRepositoryBranchesByProjectId"
         ]
       }
     },
     {
-      "label": "Description",
-      "description": "The description of the release",
+      "label": "Regular expression",
+      "description": "Return list of branches with names matching a <a href=\"https://github.com/google/re2/wiki/Syntax\" target=\"_blank\">re2</a> regular expression",
       "group": "operation",
-      "type": "Text",
+      "type": "String",
       "feel": "optional",
       "optional": true,
       "binding": {
         "type": "zeebe:input",
-        "name": "releaseDescription"
+        "name": "regex"
       },
       "condition": {
         "property": "operation",
         "oneOf": [
-          "createRelease"
+          "getRepositoryBranchesByProjectId"
         ]
       }
     },
     {
       "type": "Hidden",
-      "value": "=baseUrl + \"/api/v4/projects/\"+ projectId +\"/releases\"",
+      "value": "=baseUrl + \"/api/v4/projects/\"+ projectId +\"/repository/branches\"",
       "binding": {
         "type": "zeebe:input",
         "name": "url"
       },
       "condition": {
         "property": "operation",
         "oneOf": [
-          "createRelease"
+          "getRepositoryBranchesByProjectId",
+          "createRepositoryBranch"
         ]
       }
     },
     {
       "type": "Hidden",
-      "value": "={\"tag_name\": if releaseTagName = null then null else releaseTagName, \"name\": if releaseName = null then null else releaseName, \"description\": if releaseDescription = null then releaseDescription else releaseDescription, \"ref\": if releaseRef = null then null else releaseRef}",
+      "value": "={\"search\": if searchString = null then null else searchString, \"regex\": if regex = null then null else regex}",
       "binding": {
         "type": "zeebe:input",
-        "name": "body"
+        "name": "queryParameters"
       },
       "condition": {
         "property": "operation",
         "oneOf": [
-          "createRelease"
+          "getRepositoryBranchesByProjectId"
+        ]
+      }
+    },
+    {
+      "label": "Name",
+      "description": "Name of the branch",
```

Comment:
Nitpick: Consider adding something like "Name of the new branch" or adjusting the label to indicate this is for the branch that would be created.

My intention here is to avoid confusion with the "Reference" field.

Reply:
@wollefitz from my side, it looks like the description here is "Name of the branch", which doesn't quite fit with the comment here.

Perhaps this comment was auto-resolved accidentally? Or maybe I'm missing something.

Reply:
Never mind, I must have been looking at an old version.

I see the latest version of the file indeed says "Name of the new branch".